static-assets/components/cstudio-forms/controls/rte.js

```diff
@@ -376,70 +336,46 @@ CStudioAuthoring.Module.requireModule(
             min_height: _thisControl.rteHeight,
             theme: 'silver',
             plugins: pluginList,
-            toolbar1: toolbarConfig1,
-            toolbar2: toolbarConfig2,
-            toolbar3: toolbarConfig3,
-            toolbar4: toolbarConfig4,
             toolbar_sticky: true,
             image_advtab: true,
             encoding: 'xml',
             relative_urls: false,
             remove_script_host: false,
             convert_urls: false,
-            readonly: _thisControl.readonly,
-            force_br_newlines: forceBRTags,
-            forced_root_block: forceRootBlockPTag,
+            readonly: _thisControl.readonly, // comes from control props (not xml config)
+            force_br_newlines: forceBRTags, // comes from control props (not xml config)
+            forced_root_block: forceRootBlockPTag, // comes from control props (not xml config)
```

Comment:
Anything that comes from control props should be disallowed to be overridden by ui.xml config or merged, if applicable.